recipe/2025-09-container.md

```diff
@@ -0,0 +1,836 @@
+   - Dapr sidecar annotations (`dapr.io/enabled`, `dapr.io/app-id`, etc.)
+   - Manual scaling replica settings
+   - Kubernetes metadata (custom labels/annotations)
+
+3. **Environment Variable Processing**: Complex logic in `getEnvVarsAndSecretData()`:
+   - Connection-based environment variables with naming patterns
+   - URL parsing for connection sources  
+   - Type conversion and JSON marshaling for complex values
+   - Secret reference handling
+
+4. **Volume Dependencies**: Persistent volumes require:
+   - Dependency resolution from `options.Dependencies`
+   - Volume resource property extraction
+   - SecretProviderClass creation for Azure Key Vault
+   - Role assignment creation for volume access
+
+5. **Identity and RBAC Logic**:
+   - Conditional identity creation based on connections/volumes
+   - Azure resource naming conventions (`azrenderer.MakeResourceName()`)
+   - Role assignment scope determination
+   - Service account workload identity configuration
+
+6. **Base Manifest Merging**: Current implementation uses:
+   - `kubeutil.ParseManifest()` to deserialize YAML
+   - Object merging for Deployment, Service, ServiceAccount bases
+   - Strategic merge patch for PodSpec (`patchPodSpec()`)
+
+#### Recipe Resource Generation Using Kubernetes Extension
+
+The recipe leverages the Bicep Kubernetes extension to generate resources with native Kubernetes API schemas:
+
+**Extension Configuration:**
+```bicep
+@description('Container configuration from Radius resource')
+param containerSpec object
+@description('Kubernetes configuration')
+param kubeConfig string
+@description('Target namespace')
+param targetNamespace string = 'default'
+
+// Configure Kubernetes extension
+extension kubernetes with {
+  kubeconfig: kubeConfig
+  namespace: targetNamespace
+} as k8s
+
+// Variables for extension functionality replication
+var commonLabels = {
+  'app.kubernetes.io/name': containerSpec.name
+  'app.kubernetes.io/managed-by': 'radius'
+  'radapp.io/application': containerSpec.application ?? containerSpec.name
+}
+
+// Dapr extension support
+var daprEnabled = containerSpec.extensions?.dapr?.enabled ?? false
+var daprLabels = daprEnabled ? {
+  'dapr.io/enabled': 'true'
+  'dapr.io/app-id': containerSpec.extensions.dapr.appId ?? containerSpec.name
+} : {}
+
+var daprAnnotations = daprEnabled ? {
+  'dapr.io/enabled': 'true'
+  'dapr.io/app-id': containerSpec.extensions.dapr.appId ?? containerSpec.name
+  'dapr.io/app-port': string(containerSpec.extensions.dapr.appPort ?? 80)
+  'dapr.io/config': containerSpec.extensions.dapr.config ?? 'default'
+} : {}
+
+// Final labels combining base, extension metadata, and Dapr
+var finalLabels = union(
+  commonLabels,
+  containerSpec.extensions?.kubernetesMetadata?.labels ?? {},
+  daprLabels
+)
+
+var finalAnnotations = union(
+  containerSpec.extensions?.kubernetesMetadata?.annotations ?? {},
+  daprAnnotations
+)
```

Comment:
Note that the Kubernetes metadata configurations will move to `platformOptions` in the new container schema, see [User Story 4b](https://github.com/willtsai/design-notes-radius/blob/main/features/2025-06-compute-extensibility-feature-spec.md#user-story-4b-as-a-developer-i-want-to-define-an-application-that-can-be-deployed-to-standard-aci-confidential-aci-and-kubernetes-compute-environments-without-changing-the-core-application-logic-so-that-i-can-maintain-a-consistent-application-definition-across-different-deployment-targets)